src/screens/FilterScreen.tsx

```diff
@@ -69,84 +126,106 @@ export class FilterScreen extends React.Component<Props, State> {
-  }
-
-  handleStateUpdate = (section: any, item: any) => {
-    const { navigation } = this.props
-    const { categoryItems } = this.state
-    const value = item.value || item.id
-    const filterItems = navigation.getParam('filterItems') || []
-    const sortItems = navigation.getParam('sortItems') || []
+  getNewLocalState = async (section: any, item: any) => {
+    const { flatCategoryItems, screenName, selectedFilterItemKey } = this.state
+    const options = { flatCategoryItems, screenName } as any
 
     let stateKey
     if (section.value === PV.Filters._sectionFilterKey) {
       stateKey = 'selectedFilterItemKey'
-      const sections =
-        item.value === PV.Filters._categoryKey
-          ? sectionsWithCategory(filterItems, categoryItems, sortItems)
-          : sectionsWithoutCategory(filterItems, sortItems)
-      if (value === PV.Filters._categoryKey) {
-        this.setState({
-          sections,
-          selectedCategoryItemKey: categoryItems[0].value,
-          selectedCategorySubItemKey: '',
-          selectedFilterItemKey: value
-        })
-      } else {
-        this.setState({
-          sections,
-          selectedCategoryItemKey: '',
-          selectedCategorySubItemKey: '',
-          selectedFilterItemKey: value
-        })
+      options.selectedFilterItemKey = item.value
+      if (item.value === PV.Filters._categoryKey) {
+        const defaultCategory = await getDefaultCategory()
+        options.selectedCategoryItemKey = defaultCategory.id
       }
     } else if (section.value === PV.Filters._sectionCategoryKey) {
       if (item.parentId) {
         stateKey = 'selectedCategorySubItemKey'
-        this.setState({
-          selectedCategorySubItemKey: value
-        })
+        options.selectedCategorySubItemKey = item.value || item.id
+        options.selectedCategoryItemKey = item.parentId
       } else {
         stateKey = 'selectedCategoryItemKey'
-        this.setState({
-          selectedCategoryItemKey: value,
-          selectedCategorySubItemKey: ''
-        })
+        options.selectedCategoryItemKey = item.value || item.id
       }
+      options.selectedFilterItemKey = selectedFilterItemKey
     } else if (section.value === PV.Filters._sectionSortKey) {
       stateKey = 'selectedSortItemKey'
-      this.setState({
-        selectedSortItemKey: value
-      })
+      options.selectedSortItemKey = item.value
+      options.selectedFilterItemKey = selectedFilterItemKey
     }
 
-    return stateKey
+    const {
+      newSelectedCategoryItemKey,
+      newSelectedCategorySubItemKey,
+      newSelectedFilterItemKey,
+      newSelectedSortItemKey,
+      sections
+    } = generateSections(options)
+
+    return {
+      selectedCategoryItemKey: newSelectedCategoryItemKey,
+      selectedCategorySubItemKey: newSelectedCategorySubItemKey,
+      selectedFilterItemKey: newSelectedFilterItemKey,
+      selectedSortItemKey: newSelectedSortItemKey,
+      sections,
+      stateKey
```

Comment:
If none of your "ifs" get caught here, stateKey will be set to undefined. Is that expected?

Reply:
Looks like stateKey isn't even used anywhere 😬